infrastructure/driven-adapters/mem-stash/src/main/java/co/com/bancolombia/binstash/adapter/memory/MemoryStash.java

```diff
@@ -209,6 +211,56 @@ else if (cadidateTtl < 0 && this.expireAfter > 0) {
         return computed;
     }
 
+    @Override
+    public Mono<String> setSave(String indexKey, String key, String value, int ttl) {
+        return Mono.fromSupplier( () ->{
+            if (StringUtils.isAnyBlank(indexKey, key, value)) {
+                throw new InvalidKeyException(ERROR_KEY_MSG);
+            } else {
+                caffeineCache.put(key, new Entry(value, computeTtl(ttl)));
+                indexKeyMap.computeIfAbsent(indexKey, k ->
+                        ConcurrentHashMap.newKeySet()).add(key);
+                return value;
+            }
+        });
+    }
+
+    @Override
+    public Mono<String> setSave(String indexKey, String key, String value) {
+        return setSave(indexKey, key, value, -1);
+    }
+
+    @Override
+    public Flux<String> setGetAll(String indexKey) {
+        if (StringUtils.isAnyBlank(indexKey)) {
+            return Flux.error(new InvalidKeyException(ERROR_KEY_MSG));
+        } else {
+            Set<String> keys = indexKeyMap.getOrDefault(indexKey, Set.of());
+            return Flux.fromIterable(keys)
+                    .flatMap(key -> {
+                        Entry entry = caffeineCache.getIfPresent(key);
+                        if (entry != null ) {
+                            return Mono.just(entry.getData());
+                        } else {
+                            indexKeyMap.get(indexKey).remove(key);
```

Comment:
Although the `remove()` operation is being applied to a ConcurrentHashMap, and thread concurrency its not an issue, it is best to perform the operation within a `computeIfPresent` to guarantee an atomic operation and avoid race conditions that can ocurr between the calls to `getIfPresent` and `remove`.